portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,174 @@ function randomizeImage() {
   imageContainer.innerHTML = '';
   imageContainer.appendChild(imgElement);
 }
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addNameToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addNameToDom() function.
+  textPromise.then(addNameToDom);
+}
+
+/** Adds a name to the DOM. */
+function addNameToDom(name) {
+  console.log('Adding name to dom: ' + name);
+
+  const NameContainer = document.getElementById('name-container');
+  NameContainer.innerText = name;
+}
+
+/**
+ * Fetches comments from the servers and adds them to the DOM.
+ */
+async function getComments() {
+  await fetch('/data').then(response => response.json()).then((comments) => {
```

Comment:
using await here has no real use as you are using "then" which already waits for result. This expression will allow you to wait for some statement if you would have added something between line 69 and 70.

As seen in the below 2 jsfiddle examples (click run and look at the console at bottom in fourth quadrant) with and without await execution prints the same output.

https://jsfiddle.net/shabbirahussain/yeczpaku/3/
https://jsfiddle.net/shabbirahussain/yeczpaku/2/